cypress/integration/buy-shirt.spec.ts

```diff
@@ -1,18 +1,28 @@
+import {MenuContentPage, ShoppingCartPage, ProductsListPage} from "../page/index";
+import {LoginPage, AddressStepPage, ShippingStepPage, PaymentStepPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const shoppingCartPage = new ShoppingCartPage();
+const loginPage = new LoginPage();
+const addressSetpPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+const productsListPage = new ProductsListPage();
+
 describe("Buy a t-shirt", () => {
   it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get(".sf-menu > :nth-child(3) > a").click();
-    cy.get(".ajax_add_to_cart_button > span").click();
-    cy.get(".button-container > .button-medium > span").click();
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-    cy.get("#SubmitLogin > span").click();
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get("#cgv").click();
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get(".bankwire").click();
-    cy.get("#cart_navigation > .button > span").click();
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+    shoppingCartPage.goToAddToCardPage();
+    productsListPage.goToProdListProceedtoCheckout();
+    productsListPage.goToSummaryProceedtoCheckout();
+    loginPage.goToLogin();
+    loginPage.goToSumbitLogin();
```

Comment:
the loginPage can be simplified. It can have just one method called login and make this method to receive the email and password as parameters, in that way, the loginPage object can be used to login any user